core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -169,6 +171,19 @@ private[spark] class Executor(
 
   startDriverHeartbeater()
 
+  /**
+   * We add an empty WebUI in executor to enable Executor MetricsServlet sink if needed.
+   */
+  private val executorWebUiEnabled = conf.getBoolean("spark.executor.ui.enabled", false)
+  private[executor] var webUi: ExecutorWebUI = _
+  if (executorWebUiEnabled && !isLocal) {
+     webUi = new ExecutorWebUI(conf, env.securityManager, SparkUI.getUIPort(conf))
+     webUi.bind()
+     env.metricsSystem.getServletHandlers.foreach(webUi.attachHandler)
+     heartbeatReceiverRef.ask[Boolean](ReportExecutorWebUrl(executorId, webUi.webUrl))
+     logInfo(s"Starting executor web ui at ${webUi.webUrl}")
```

Comment:
It is too overkill to start an jetty server on each executor only for metrics. I believe you have many different ways to collect executor metrics other than servlet. 

Besides, to get executor URL you add a new heartbeat message, basically I think it is too overkill.

Reply:
Thanks @jerryshao . It's overkill for this purpose. But it offers a pull way to get executor metrics. I will think twice how to keep a pull way to get that.

Reply:
Maybe you can use jmx or some other tools. Basically the metrics is still report in every N seconds, so pulling frequently doesn't help increasing the accuracy.